crates/sparrow-syntax/src/syntax/signature.rs

```diff
@@ -1,7 +1,53 @@
+use hashbrown::HashSet;
 use itertools::Itertools;
 
 use crate::parser::try_parse_signature;
-use crate::{ExprRef, FeatureSetPart, FenlType, Located, Parameters, ParseErrors};
+use crate::{ExprRef, FeatureSetPart, FenlType, Located, Parameters, ParseErrors, TypeConstraint};
+
+/// Type variable defined by a signature
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+#[cfg_attr(test, derive(serde::Serialize))]
+pub struct TypeVariable(pub String);
+
+impl std::fmt::Display for TypeVariable {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+/// Contains the type variable and the constraints for
+/// that variable.
+///
+/// e.g. N(type variable): Eq(constraint) + Hash(constraint)
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+#[cfg_attr(test, derive(serde::Serialize))]
+pub struct TypeParameter {
+    pub name: TypeVariable,
+    pub constraints: Vec<TypeConstraint>,
```

Comment:
Looks like we already support multiple constraints?

Reply:
We allow the ability to have multiple here, but we don't have the logic for finding the LUB for multiple at the moment. That will come in a follow up